packages/tooltips/src/styled/StyledTooltip.ts

```diff
@@ -5,78 +5,178 @@
+import { StyledParagraph } from './StyledParagraph';
+import { StyledTitle } from './StyledTitle';
+import { arrowStyles } from './arrow-styles';
 
 const COMPONENT_ID = 'tooltip.tooltip';
 
 export type TOOLTIP_SIZE = 'small' | 'medium' | 'large' | 'extra-large';
+export type TOOLTIP_TYPE = 'light' | 'dark';
 
-const shouldShowArrow = ({ hasArrow, placement }: IStyledTooltipProps) => {
-  return hasArrow && placement;
-};
+export interface IStyledTooltipProps {
+  hasArrow?: boolean;
+  size?: TOOLTIP_SIZE;
+  /** All valid [Popper.JS Placements](https://popper.js.org/popper-documentation.html#Popper.placements) */
+  placement: POPPER_PLACEMENT;
+  zIndex?: number | string;
+  type: TOOLTIP_TYPE;
+}
+
+const sizeStyles = ({
+  theme,
+  size,
+  hasArrow,
+  type
+}: IStyledTooltipProps & ThemeProps<DefaultTheme>) => {
+  let margin = `${theme.space.base * 1.5}px`;
+  let borderRadius = theme.borderRadii.sm;
+  let padding = '0 1em';
+  let maxWidth;
+  let overflowWrap;
+  let whiteSpace = 'nowrap';
+  let lineHeight = (theme.space.base * 5) / stripUnit(theme.fontSizes.sm);
+  let fontSize = theme.fontSizes.sm;
+  let titleDisplay;
+  let paragraphMarginTop;
+  let wordWrap;
+  let arrowFontSize = `${theme.space.base * 1.75}px`;
+
+  if (size !== 'small') {
+    borderRadius = theme.borderRadii.md;
+    overflowWrap = 'break-word';
+    whiteSpace = 'normal';
+    wordWrap = 'break-word';
+  }
+
+  if (size === 'extra-large') {
+    padding = `${theme.space.base * 10}px`;
+    maxWidth = `460px`;
+    lineHeight = (theme.space.base * 5) / stripUnit(theme.fontSizes.md);
+    paragraphMarginTop = `${theme.space.base * 2.5}px`;
+    arrowFontSize = `${theme.space.base * 4}px`;
+  } else if (size === 'large') {
+    padding = `${theme.space.base * 5}px`;
+    maxWidth = `270px`;
+    lineHeight = (theme.space.base * 5) / stripUnit(theme.fontSizes.md);
+    paragraphMarginTop = `${theme.space.base * 2}px`;
+    arrowFontSize = theme.fontSizes.sm;
+  } else if (size === 'medium') {
+    padding = '1em';
+    maxWidth = `140px`;
+    lineHeight = (theme.space.base * 4) / stripUnit(theme.fontSizes.sm);
+  }
+
+  if (type === 'light') {
+    arrowFontSize = math(`${arrowFontSize} + 2px`);
+  }
+
+  if (size === 'extra-large' || size === 'large') {
+    fontSize = theme.fontSizes.md;
+    titleDisplay = 'block';
+  }
 
-const retrieveTooltipMargin = ({ hasArrow, size }: IStyledTooltipProps) => {
   if (hasArrow) {
     if (size === 'large') {
-      return 'margin: 8px;';
+      margin = `${theme.space.base * 2}px`;
     } else if (size === 'extra-large') {
-      return 'margin: 12px;';
+      margin = `${theme.space.base * 3}px`;
     }
   }
 
-  return 'margin: 6px';
+  return css`
+    margin: ${margin};
+    border-radius: ${borderRadius};
+    padding: ${padding};
+    max-width: ${maxWidth};
+    line-height: ${lineHeight};
+    word-wrap: ${wordWrap};
+    white-space: ${whiteSpace};
+    /*  Manipulate arrow sizing (which is based on ems). */
```

Comment:
If I'm reading it right, I think this comment might be stale. Arrow sizing is manipulated via the `arrowFontSize` applied to the `::before/::after` below.

Reply:
I brought this comment over from https://github.com/zendeskgarden/css-components/blob/master/packages/tooltips/src/_base.css#L48-L53

I think it was originally implying that the font-size is impacting the `::before,::after` styling in `c-arrow`. This is still based on `em`'s. We are using

```ts
const arrowSize = `${stripUnit(theme.fontSizes.sm) / stripUnit(theme.fontSizes.md)}em`;
```

in `arrow-styles` right now.

Should I just shorten this to `/* Manipulate arrow sizing */`?